packages/clerk-js/src/core/clerk.ts

```diff
@@ -708,8 +708,16 @@ export class Clerk implements ClerkInterface {
     // to keep the organization id that the session had.
     const shouldSwitchOrganization = organization !== undefined;
     if (newSession && shouldSwitchOrganization) {
-      const organizationId = typeof organization === 'string' ? organization : organization?.id;
-      newSession.lastActiveOrganizationId = organizationId || null;
+      const organizationIdOrSlug = typeof organization === 'string' ? organization : organization?.id;
+
+      if (organizationIdOrSlug?.startsWith('org_')) {
+        newSession.lastActiveOrganizationId = organizationIdOrSlug || null;
+      } else {
+        const matchingOrganization = session?.user?.organizationMemberships.find(
```

Comment:
Worth noting that this won't work if the slug is for an organization that they're not a member of, but the backend would reject that case anyway!

On reflection, if we fail to find an organization here (maybe because the user isn't a member), we might want to throw a helpful error here.

Reply:
Also: what happens here if session.user is null, and under what circumstances would that happen?

Reply:
> Also: what happens here if session.user is null, and under what circumstances would that happen?

This is a good catch! We have a condition in the beginning of the `setActive` method that checks if there's an available session already, and we should be using the `session` property from the `Clerk` class instead of the local one which will always have a value when it reaches the organization ID check:

https://github.com/clerk/javascript/blob/bb2e8720e8f6bb955d4f100d4f998065fec52727/packages/clerk-js/src/core/clerk.ts#L681-L685

So instead of `session?.user` we should be doing `this.session?.user` instead. Thanks!

Reply:
Will add tests!